lib/server/index.js

```diff
@@ -223,6 +227,11 @@ Server.prototype = {
     var config = this.config;
     var test_page = config.get('test_page');
 
+    // If there is an array of test pages, we cycle through them as the default
+    if (Array.isArray(test_page)) {
+      test_page = test_page[this.testPageCount % test_page.length];
+    }
```

Comment:
So I attempted number 2, but am still running into failures. Could it be due to the fact that runners are sharing an instance of a launcher?

It's a bit hard for me to grok all that happens, but it seems like maybe using the same launcher more than once can clobber the socket communications and wind up causing failures. So, rather than creating multiple instances of runners, it looks like maybe I need to create multiple launcher instances where each gets it's own runner, does that sound logical?